src/bit-systems/pdf-menu-system.ts

```diff
@@ -0,0 +1,99 @@
+import { defineQuery, entityExists, hasComponent } from "bitecs";
+import { Text } from "troika-three-text";
+import type { HubsWorld } from "../app";
+import { HoveredRemoteRight, Interacted, MediaPDF, NetworkedPDF, PDFMenu } from "../bit-components";
+import { anyEntityWith, findAncestorWithComponent } from "../utils/bit-utils";
+import type { EntityID } from "../utils/networking-types";
+import { takeOwnership } from "../utils/take-ownership";
+import { setMatrixWorld } from "../utils/three-utils";
+import { PDFComponentMap } from "./pdf-system";
+
+function clicked(world: HubsWorld, eid: EntityID) {
+  return hasComponent(world, Interacted, eid);
+}
+
+function findPDFMenuTarget(world: HubsWorld, menu: EntityID, sceneIsFrozen: boolean) {
+  if (sceneIsFrozen) {
+    PDFMenu.targetRef[menu] = 0;
+    PDFMenu.clearTargetTimer[menu] = 0;
+    return;
+  }
+
+  const hovered = hoveredQuery(world);
+  const target = hovered.map(eid => findAncestorWithComponent(world, MediaPDF, eid))[0] || 0;
```

Comment:
will just keep pointing out that I am very worried about using map/filter/etc in code called every frame. Wish I had a solid way of proving to myself that this isn't a problem...